# **Capstone Project - The Best Neighborhood to Live In**
## **Table of Content**


1.   Introduction: Business Problem
2.   Data
3.   Methodology
4.   Results and Discussion
5.   Conclusion



## **Introduction: Business Problem**

When people are looking for a change of apartment, one of the most important
deciding factors is the neighborhood.
*   Some people might be on a budget while some might be looking for something luxurious
*   A few families would prefer having schools nearby
*   While, some just might be foodies who would love to have many restaurants near their house

We will use our data science abilities to predict the most appropriate neighborhoods for people with the above mentioned preferences.

Here, the **target audience would be people looking for apartments.**

The factors which influence our problem are:


1.   Number of schools in each neighborhood
2.   Number of restaurants in each neighborhood
3.   Budget of prospective tenant






## **Data**
Following data sources have been used to extract/generate the required information:

1.Foursquare API data: to obtain details of restaurants, schools etc. in each neighborhood

 2.Dataset from Kaggle:https://catalog.data.gov/dataset/new-york-city-population-by-neighborhood-tabulation-areas-f2447 
 
 3.https://www.kaggle.com/jackcook/neighborhoods-in-new-york/data : Will be using to obtain population statistics of neighborhood in New York
 
 4.https://data.cityofnewyork.us/City-Government/Neighborhood-Tabulation-Areas-NTA-/cpf4-rkhq : To obtain geospatial data of neighborhood in New York.

5.https://data.cityofnewyork.us/Housing-Development/Housing-New-York-Units-by-Building/hg8x-zxpr : To obtain information regarding housing units in New York


Installing the required libraries

In [2]:
import requests # library to handle requests
import pandas as pd # library for data analsysis
import numpy as np # library to handle data in a vectorized manner
import random # library for random number generation

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values
!conda install -c conda-forge folium=0.5.0 --yes
import folium

print('Folium installed and imported!')

/bin/bash: conda: command not found
/bin/bash: conda: command not found
Folium installed and imported!


Getting Coordinates of New York

In [4]:
address = 'New York, NY'

geolocator = Nominatim(user_agent="foursquare_agent")
geolocator = Nominatim(user_agent='foursquare_agent')
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The coordinates of New York are',latitude ,longitude)

The coordinates of New York are 40.7127281 -74.0060152


The Map of New York

In [6]:
nyc_map = folium.Map(location=[latitude,longitude], zoom_start=12 )
nyc_map

In [0]:
#Dataset containing details about size of each neighborhood
df = pd.read_csv('nynta.csv')

In [0]:
df.head()

BoroCode   BoroName  ...    Shape_Leng    Shape_Area
0         3   Brooklyn  ...  39247.227831  5.400502e+07
1         4     Queens  ...  33266.904797  5.248828e+07
2         4     Queens  ...  19816.712074  1.972685e+07
3         4     Queens  ...  20976.335574  2.288777e+07
4         1  Manhattan  ...  17040.685413  1.064708e+07

[5 rows x 8 columns]

In [0]:
df['Shape_Area'].max()
print(df.loc[df.Shape_Area == 327768572.857,['NTAName','Shape_Area']])

                               NTAName    Shape_Area
176  New Springville-Bloomfield-Travis  3.277686e+08


**Largest Neighborhood in New York** : New Springville-Bloomfield-Travis

Population Statistics for Each Neighborhood

In [0]:
df_pop = pd.read_csv('NYC pop.csv')
df_pop.head()

Borough  Year  ...                         NTA Name Population
0   Bronx  2000  ...               Claremont-Bathgate      28149
1   Bronx  2000  ...  Eastchester-Edenwald-Baychester      35422
2   Bronx  2000  ...       Bedford Park-Fordham North      55329
3   Bronx  2000  ...                          Belmont      25967
4   Bronx  2000  ...                        Bronxdale      34309

[5 rows x 6 columns]

In [0]:
#Neighborhood with maximum population
df_pop['Population'].max()

136954

In [0]:
print(df_pop.loc[df_pop.Population==136954, ['NTA Name','Population']])

           NTA Name  Population
95  Upper West Side      136954


**Neighborhood in New York with the maximum population** : Upper West Side

### **Using FourSquare API**

In [0]:
CLIENT_ID = '1RJCHNW45W2WNLEBWOF2KULSEXBFSIADLFSBAMDCOVADRGV4' 
CLIENT_SECRET = '2CK3RNSRP4HYE1QI5PT0LMDGCPYFBC2EAX1ESZ4V3HOC5WSC' 
VERSION = '20180604'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 1RJCHNW45W2WNLEBWOF2KULSEXBFSIADLFSBAMDCOVADRGV4
CLIENT_SECRET:2CK3RNSRP4HYE1QI5PT0LMDGCPYFBC2EAX1ESZ4V3HOC5WSC


Schools in New York

In [0]:
search_query = 'School'
radius = 500
print(search_query + ' .... OK!')


School .... OK!


In [0]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=1RJCHNW45W2WNLEBWOF2KULSEXBFSIADLFSBAMDCOVADRGV4&client_secret=2CK3RNSRP4HYE1QI5PT0LMDGCPYFBC2EAX1ESZ4V3HOC5WSC&ll=40.7127281,-74.0060152&v=20180604&query=School&radius=500&limit=30'

In [0]:
results = requests.get(url).json()
venues = results['response']['venues']


dataframe = json_normalize(venues)
dataframe.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:5: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  """


id  ... venuePage.id
0  4b980f0ef964a520952935e3  ...          NaN
1  4d5ab621e2df60fc9f09d4e5  ...     32694453
2  4e37fc7918a8470916d00b74  ...          NaN
3  4dbc221c6a23e294ba25a512  ...          NaN
4  4a78d55cf964a5208be61fe3  ...          NaN

[5 rows x 18 columns]

In [0]:
dataframe.shape[0]

30

There are a total of 30 schools in New York.

In [0]:
df_pc = dataframe['location.postalCode']
df_pc

0     10038
1     10038
2     10038
3     10013
4     10279
5     10038
6     10013
7     10038
8     10038
9       NaN
10    10038
11    10007
12      NaN
13    10007
14      NaN
15    10007
16    10007
17      NaN
18    10007
19    10007
20    10038
21    10038
22      NaN
23    10038
24    10038
25    10038
26    10013
27    10013
28    10013
29    10013
Name: location.postalCode, dtype: object

In [0]:
df_pc.value_counts()

10038    12
10013     6
10007     6
10279     1
Name: location.postalCode, dtype: int64

This indicates that the area which has the Postal Code 10038 is a suitable option for those buyers/tenants whose kids go to school.
The neighborhoods which have the Postal Code 10038 are :


1.   City Hall
2.   Manhattan
3.   Downtown





In [15]:
data1 = [['City Hall',40.7128,-74.0061],['Manhattan',40.7831,-73.9712],['Downtown',40.7209,-74.0007]] 
df2 = pd.DataFrame(data1, columns=['Neighborhood','Lat','Long'])
df2.head()

Neighborhood      Lat     Long
0    City Hall  40.7128 -74.0061
1    Manhattan  40.7831 -73.9712
2     Downtown  40.7209 -74.0007

Map Showing Top 3 Neighborhoods with **Highest Number of Schools**

In [26]:
r = folium.Map(location=[latitude,longitude], zoom_start=12)
tip = 'Click me!'

folium.Marker([40.7128, -74.0061], popup='<b>City Hall</b>', tooltip=tip).add_to(r)
folium.Marker([40.7831, -73.9712], popup='<b>Manhattan</b>', tooltip=tip).add_to(r)
folium.Marker([40.7209, -74.0007], popup='<b>Downtown</b>', tooltip=tip).add_to(r)


r

Next, we look for neighborhood with a lot of restaurants nearby.

In [0]:
search_query2 = 'restaurant'
radius = 500
print(search_query2 + ' .... OK!')

restaurant .... OK!


In [0]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url
results = requests.get(url).json()
venues = results['response']['venues']

df_r = json_normalize(venues)
df_r.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  


id  ... venuePage.id
0  4bc238adf8219c744286b410  ...          NaN
1  4b4fdfc8f964a520801827e3  ...    131643631
2  4c4890da1879c9b6cce6e143  ...          NaN
3  4b88199af964a520e6df31e3  ...          NaN
4  45e5c256f964a52046431fe3  ...     43156651

[5 rows x 24 columns]

In [0]:
df_r.shape[0]

30

There are 30 restaurants in New York.

In [0]:
df_r['location.postalCode'].value_counts()

10007    15
10038     8
10013     5
10004     2
Name: location.postalCode, dtype: int64

In [11]:
data = [['Tribeca', 40.7163,-74.0086], ['Downtown', 40.7209,-74.0007], ['Lower Manhattan', 40.7549,-73.9840]]
df_rest = pd.DataFrame(data, columns = ['Neighborhood', 'Lat','Long'])
df_rest.head() 
  

Neighborhood      Lat     Long
0          Tribeca  40.7163 -74.0086
1         Downtown  40.7209 -74.0007
2  Lower Manhattan  40.7549 -73.9840

This implies people who prefer eating out alot should look for apartments in the neighborhoods with postal code 10007. The neighborhoods which have 10007 postal code are : 


1.   Tribeca
2.   Downtown
3.   Lower Manhattan



Map displaying Top 3 Neighborhoods with highest **Number of Restaurants**

In [18]:
m = folium.Map(
    location=[latitude,longitude],
    zoom_start=12,
    
)

tooltip = 'Click me!'

folium.Marker([40.7549, -73.9840], popup='<b>Lower Manhattan</b>', tooltip=tooltip).add_to(m)
folium.Marker([40.7163, -74.0086], popup='<b>Tribeca</b>', tooltip=tooltip).add_to(m)
folium.Marker([40.7209, -74.0007], popup='<b>Downtown</b>', tooltip=tooltip).add_to(m)


m

Classifying the neighborhood on the basis of how many **Extremely Low Income Units and Middle Income Units** they have.

In [0]:
df_b = pd.read_csv('Housing_New_York_Units_by_Building.csv')
df_b.head()

Project ID  ... Total Units
0       64390  ...           4
1       64390  ...          10
2       64390  ...          16
3       64390  ...          17
4       64390  ...          18

[5 rows x 42 columns]

In [0]:
df_l = df_b['Extremely Low Income Units']
df_l

0        0
1        0
2       15
3       14
4       17
        ..
4546     0
4547     0
4548     0
4549     0
4550     0
Name: Extremely Low Income Units, Length: 4551, dtype: int64

In [0]:
df_l.sort_values(ascending=False,axis=0)

3891    429
1031    250
1490    247
1379    241
486     239
       ... 
3004      0
3003      0
3002      0
3000      0
0         0
Name: Extremely Low Income Units, Length: 4551, dtype: int64

Top 5 neighborhoods with maximum number of extremely low income housing units

In [0]:
df_b.iloc[[3891,1031,1490,1379,486]]

Project ID  ... Total Units
3891       55396  ...         536
1031       65341  ...         259
1490       62209  ...         292
1379       67426  ...         405
486        68177  ...         278

[5 rows x 42 columns]

Hence, people who are **tight on a budget** should look for apartments in these 5 neighborhoods.

Map displaying Top 5 neighborhoods with highest number of **Extremely Low Income Units**

In [27]:
e = folium.Map(
    location=[latitude,longitude],
    zoom_start=10,
    
)

tooltip = 'Click me!'

folium.Marker([40.63910,-74.170235], popup='<b>Annadale-Huguenot-Princes Bay-Eltingville</b>', tooltip=tooltip).add_to(e)
folium.Marker([40.574821,-74.001044], popup='<b>Seagate-Coney Island</b>', tooltip=tooltip).add_to(e)
folium.Marker([40.856665,-73.875798], popup='<b>park-cemetery-etc-Bronx</b>', tooltip=tooltip).add_to(e)
folium.Marker([40.782965,-73.950061], popup='<b>Yorkville</b>', tooltip=tooltip).add_to(e)
folium.Marker([40.762505,-73.985936], popup='<b>Midtown-Midtown South</b>', tooltip=tooltip).add_to(e)
e

In [0]:
df_m = df_b['Middle Income Units']
df_m.sort_values(ascending=False, axis=0)

2466    448
4267    422
2482    352
4082    343
1010    284
       ... 
2896      0
2894      0
2893      0
2887      0
0         0
Name: Middle Income Units, Length: 4551, dtype: int64

Top 5 neighborhood with **maximum number of Middle Income Housing Units**

In [0]:
df_b.iloc[[2466,4267,2482,4082,1010]]

Project ID  ... Total Units
2466       52774  ...         539
4267       55506  ...         422
2482       63371  ...         352
4082       55511  ...         343
1010       61931  ...         800

[5 rows x 42 columns]

Therefore, people who are **willing to pay above than average rent**, should look for apartments in these 5 neighbourhoods.

Map displaying Top 5 Neighborhoods with Highest Number of **Middle Income Housing Units**

In [28]:
i = folium.Map(
    location=[latitude,longitude],
    zoom_start=10,
    
)

tooltip = 'Click me!'

folium.Marker([40.700836,-73.806551], popup='<b>Hammels-Arverne-Edgemere</b>', tooltip=tooltip).add_to(i)
folium.Marker([40.697436,-73.992881], popup='<b>Brooklyn</b>', tooltip=tooltip).add_to(i)
folium.Marker([40.877190,-73.889569], popup='<b>Bronxdale</b>', tooltip=tooltip).add_to(i)
folium.Marker([40.593738,-73.787140], popup='<b>Queens</b>', tooltip=tooltip).add_to(i)
folium.Marker([40.742166,-73.960033], popup='<b>Springfield Gardens North</b>', tooltip=tooltip).add_to(i)
i

## **Methodology**

In this project, our main aim is to shortlist neighborhoods as per the requirements of the prospective tenants. Some of the classifying factors include the rent, the number of schools, the number of restaurants etc.

Using Foursquare API data, we have first found the **total number of schools and restaurants** in each neighborhood in New York. 

Then, we have shortlisted the areas having the maximum number of schools and restaurants **based on their postal codes in order to maintain accuracy**.

Next, we have performed analysis on the NYC Housing Data and classified neighborhoods on the basis of how many extremely low income units and how many middle income units they have.

Finally, we have **represented all these shortlisted locations on the New York map according to each criteria respectively**.




## **Results and Discussion**

Upon analysing all the datasets we observe that there many neighborhoods in New York, each with its pros and cons. 

Neighborhood with the **largest population**: Upper West Side

Neighborhood with the **largest area**: New Springville-Bloomfield-Travis

For **people on a budget**, the following neighborhoods are best suited:



1.   Annadale-Huguenot-Princes Bay-Eltingville
2.   Seagate-Coney Island
3.   Park-cemetery-etc-Bronx
4.   Yorkville
5.   Midtown-Midtown South


Prospective tenants **willing to pay above average rents** should look for an apartment in the following neighborhoods:



1.   Hammels-Arverne-Edgemere
2.   Brooklyn
3.   Bronxdale
4.   Queens
5.   Springfield Gardens North




We see that for **prospective tenants with kids**, the following 3 neighborhoods are the best suited:


1.   City Hall
2.   Manhattan
3.   Downtown

For people looking for an apartment **with alot of restaurants nearby**, the following neighborhoods are best suited:



1.   Tribeca
2.   Lower Manhattan
3.   Downtown







## **Conclusion**

The purpose of this project is to help prospective tenants find the right neighborhood for themselves. 

We have shortlisted a few neighborhoods according to the 3 criterias; schools, restaurants and budget. There are many more classifying criteria which can be used if the datasets for the same are available.

This is **not an exhaustive analysis** as the definition of the Best Neighborhood to Live In is different for every individual and the data for every criteria is not accessible. 

The final decision on an optimal neighborhood can only be made by the prospective tenant after taking into consideration other factors such as proximity to public transport facilities, healthcare facilities, cleanliness, climate etc.

This analysis will surely help the prospective tenants by giving them a rough outline and **a place to start their search of an apartment from**.

